### PR TITLE
bean post processor 

### PR DESCRIPTION
**Pull request by Wectro20**

### CODE REVIEW

File: src/main/kotlin/com/ajax/cryptocurrency/bpp/ScheduledBackgroundJobBeanPostProcessor.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            beans[beanName] = beanClass
+        }
+        return bean
+    }
+
+    override fun postProcessAfterInitialization(bean: Any, beanName: String): Any {
+        val originalBean = beans[beanName]
+        if (originalBean != null) {
+            val annotation = originalBean.java.getAnnotation(ScheduledBackgroundJobStarter::class.java)
+            val scheduler = Executors.newScheduledThreadPool(cryptocurrencyNames.size)
```
Comment (ajax-gnatiuk-s) [inline src/main/kotlin/com/ajax/cryptocurrency/bpp/ScheduledBackgroundJobBeanPostProcessor.kt:32]: Sorry, forgot about one more thing.

Suppose you have multiple beans that are annotated with `ScheduledBackgroundJobStarter`.
So, for each such bean you create personal executor service.

It's better to use common executor for all such beans.
1. declare scheduler as property of the BPP
```kotlin
class ScheduledBackgroundJobBeanPostProcessor(...) {
    private val scheduler = Executors.newScheduledThreadPool(cryptocurrencyNames.size)
    
    ...
}    
```
2. define pre destroy function to shutdown the executor:
```kotlin
@PreDestroy
private fun shutdown() {
   executor.shutdownNow()
}
```

File: src/main/kotlin/com/ajax/cryptocurrency/parser/interfaces/ParserInterface.kt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package com.ajax.cryptocurrency.parser.interfaces
+
+fun interface ParserInterface {
+    fun priceSaver(cryptocurrencyName: String)
```
Comment (ajax-gnatiuk-s) [inline src/main/kotlin/com/ajax/cryptocurrency/parser/interfaces/ParserInterface.kt:4]: usually, we name a function  or method with verb. But `price saver` is noun

**Review by ajax-gnatiuk-s — APPROVED** [review]
It was the final review in this PR. Let's fix and merge. Good work!

File: src/main/kotlin/com/ajax/cryptocurrency/bpp/ScheduledBackgroundJobBeanPostProcessor.kt (hunk 1 above)
Comment (Wectro20) [reply src/main/kotlin/com/ajax/cryptocurrency/bpp/ScheduledBackgroundJobBeanPostProcessor.kt:32]: done
